text/0000-url-primitives.md

```diff
@@ -0,0 +1,352 @@
+- Start Date: 2020-01-04
+- Relevant Team(s): Ember.js
+- RFC PR: https://github.com/emberjs/rfcs/pull/570
+- Tracking: (leave this empty)
+
+# URL Manager
+
+Supersedes [Add queryParams to the router service](https://github.com/emberjs/rfcs/pull/380)
+
+## Summary
+
+A URL Manager will provide the ability to manipulate the URL, 
+both when writing to `window.location` (serializing), 
+and reading from `window.location` (deserializing). 
+The ultimate goal of the URL Manager will be to enable 
+app and addon authors to customize the use of the URL—
+including internationalized routes, dynamic segment slugs / aliases, 
+and alternative query params behavior—though, 
+implementation of those specific things is outside the scope of *this* RFC. 
```

Comment:
isn't that something that the `Location` API can already handle? 🤔

Reply:
Location api can't handle nested query params, which is a common ask